src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -895,6 +895,19 @@ def create_express_route_peering(
     """
     from azure.mgmt.network.models import \
         (ExpressRouteCircuitPeering, ExpressRouteCircuitPeeringConfig)
+
+    # TODO: Remove workaround when issue #1574 is fixed in the service
+    # region Issue #1574 workaround
+    circuit = _network_client_factory().express_route_circuits.get(
+        resource_group_name, circuit_name)
+    if peering_type == 'MicrosoftPeering' and circuit.sku.tier.lower() == 'standard':
+        raise CLIError("MicrosoftPeering cannot be created on a 'Standard' SKU circuit")
+    for peering in circuit.peerings:
+        if str(peering.vlan_id) == vlan_id:
```

Comment:
If the `vlan_id` is not digital only, then case insensitive comparison would be more reliable